test/sys/test_signal.rs

```diff
@@ -99,9 +109,6 @@ fn test_signal() {
     assert!(SIGNALED.load(Ordering::Relaxed));
     assert_eq!(unsafe { signal(Signal::SIGINT, SigHandler::SigDfl) }.unwrap(), handler);
 
-    let action_handler = SigHandler::SigAction(test_sigaction_action);
```

Comment:
Ok, this was dumb on our part to have all of that different functionality in a single test.  Thanks for splitting it up a bit.